sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSessionManager.java

```diff
@@ -247,7 +256,10 @@ private AmqpErrorContext getErrorContext() {
      * @return A Mono that completes when an unnamed session becomes available.
      * @throws AmqpException if the session manager is already disposed.
      */
-    private Mono<ServiceBusReceiveLink> getActiveLink() {
+    Mono<ServiceBusReceiveLink> getActiveLink() {
+        if (this.receiveLink != null) {
+            return Mono.just(this.receiveLink);
```

Comment:
Once user have accepted a session, their receiver is locked to one session (named or unnamed) . And if they call receiver.receiveMessages() two times parallelly , what should happen ? There will be two subscriber receiving from same link. These two subscriber will get duplicated messages. We can check in dotnet, if they are throwing exception while one subscriber is receiving and do not allow second receiveMessages () ?

Reply:
According to our offline discussion, we give the user full control over how many times they subscribe. But we return the same Flux instance if they call the async receiveMessages() twice.